book/basics/units_basics.nim

```diff
@@ -0,0 +1,150 @@
+of the work at compile-time so that you know that if the code compiles, then the units are correct. 
+Other libraries would probably check this at runtime, and it will complain first when it sees the piece of code it wants to check
+the units of. As an added bonus, doing most of the work when compiling the code will make the code run faster as there will at most
+be a conversion factor injected.
+
+## Motivating example
+
+One example of a real disaster caused by conversion errors is the Mars Climate Orbiter that NASA sent to the red planet in 1999. 
+It crashed. Why? Because the navigators at JPL used metric units while the manufacturers had used imperial units. So when
+the navigators thought they read 1 N⋅s from the craft's sensors it was in fact 1 lbf⋅s.
+
+As an introduction to `Unchained`, let's find out what 1 lbf⋅s is in N⋅s to understand how much of a difference it made.
+"""
+nbCode:
+  import unchained
+
+  ## Define composite units
+  defUnit(N•s)
+  defUnit(lbf•s)
+
+  ## Assign variables
+  let lbfs = 1.lbf•s
+  let Ns = lbfs.to(N•s)
+
+  echo lbfs, " equals ", Ns
+
+nbText: """
+As you can see, they thought the value they read was 4.4 times smaller than it really was! 
+
+But what does the code do? Let's dissect it block-by-block:
+
+1. `defUnit(N•s)` - There are a ridiculous amount of combinations of base units.
+Therefore you must define composite units manually before you use them. More on this in a later section.
+2. `let lbfs = 1.lbf•s` - We use the dot (`.`) to assign a unit to a number. In this case we assign a variable with the value "1 lbf•s".
+3. `let Ns = lbfs.to(N•s)` - We use the `to` template to convert a variable from one unit to another.
+"""
+
+nbText: """
+## Defining and assigning units
+
+The first thing you probably will want to do is to associate a value with a unit, for example 1 Newton. There are multiple ways
+to specify a unit, both short and long versions:
+"""
+
+nbCodeInBlock:
+  ## Short version
+  let n1 = 1.N
+  ## Long version
+  let n2 = 1.Newton
+  ## Short version with prefix (milli)
+  let n3 = 1.mN
+  ## Long version with prefix
+  let n4 = 1.MilliNewton
+
+nbText: """
+That was easy enough, and being able to use prefixes makes for a saner user experience.
+When it comes to composite types it gets a bit more complicated though. For example if we want to use a unit `kg•m•s⁻¹` we
+have a few more things to consider. First and foremost, composite types must either be defined in a `defUnit` or be used in a
+dot expression (eg `10.kg•m•s⁻¹`) *before* it can be used in other parts of the code. Here comes a few valid and invalid cases:
+"""
+
+nbCodeInBlock:
+  ## Correct way of doing it!
+  defUnit(kg•m•s⁻¹)
+  proc unitProc(k: kg•m•s⁻¹) =
+    echo k
+
+nbText: """
+  ```nim
+  ## Incorrect way of doing it! Missing `defUnit`
+  proc unitProc(k: kg•m•s⁻¹) =
+    echo k
+  ```
+"""
+
+#[
```

Comment:
Is this comment still here for a reason?

Reply:
It's commented out because it didn't work. When it does, I will uncomment it 😉 was something with Unchained only creating the long version when using the `.`. Or is it already fixed? 😅

Reply:
Ah, indeed. I misread the comment. I thought it was the same code as the one in use above.

No, that is not fixed yet.